shared/Microsoft.Extensions.TaskCache.Sources/TaskCache.cs

```diff
@@ -14,10 +14,10 @@ internal static class TaskCache
         /// We're caching this in a static readonly field to make it more inlinable and avoid the volatile lookup done
         /// by <c>Task.CompletedTask</c>.
         /// </remarks>
-#if NET451
+#if NET451 || NET452
```

Comment:
you can do #if !NETSTANDARD*, so each assembly would not have own completed task instance at least on core.